ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/StartupUtil.java

```diff
@@ -107,18 +107,24 @@ private static Eth1Data get_eth1_data_stub(BeaconState state, UnsignedLong curre
   }
 
   public static BeaconStateWithCache createInitialBeaconState(ArtemisConfiguration config) {
+    BeaconStateWithCache initialState;
+    UnsignedLong genesisTime = UnsignedLong.valueOf(config.getGenesisTime());
     if (config.getInteropActive()) {
-      return createMockedStartInitialBeaconState(config);
+      initialState = createMockedStartInitialBeaconState(config);
+    } else {
+      initialState =
+          BeaconStateUtil.initialize_beacon_state_from_eth1_new(
+              Bytes32.ZERO,
+              UnsignedLong.valueOf(config.getGenesisTime()),
+              newDeposits(config.getNumValidators()));
     }
-    return BeaconStateUtil.initialize_beacon_state_from_eth1_new(
-        Bytes32.ZERO,
-        UnsignedLong.valueOf(config.getInteropGenesisTime()),
-        newDeposits(config.getNumValidators()));
+    initialState.setGenesis_time(genesisTime);
```

Comment:
It seems to me that line 121 should be in the if block instead of the else block. Because in interop, yes we do want to explicity set the time, in not interop mode, we dont want to stray any state variable from spec

Reply:
in non interop mode it is still a simulation.  the only real mode is through the deposit event kicking off genesis.  I tracked down those rabbit holes, but happy to walk u thru what I found

Reply:
in other words, I want to be able to specify the genesis time explicitly unless we are doing a run starting from deposits

Reply:
Yea that makes sense, and its way cleaner than before. Good job.